hibernate-core/src/main/java/org/hibernate/engine/spi/EnhancedEntity.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.engine.spi;
+
+/**
+ * This is a special marker interface designed to represent the union of several traits:
+ * - ManagedEntity
+ * - Managed
+ * - PersistentAttributeInterceptable
+ * - SelfDirtinessTracker
+ * The need for such a "union" isn't natural in the Java language, but represents a technicality
+ * we need to bypass performance issues caused by https://bugs.openjdk.org/browse/JDK-8180450
+ * @see org.hibernate.engine.internal.ManagedTypeHelper
+ * @see org.hibernate.engine.spi.Managed
+ * @see org.hibernate.engine.spi.ManagedEntity
+ * @see PersistentAttributeInterceptable
+ * @see SelfDirtinessTracker
+ */
+public interface EnhancedEntity extends ManagedEntity, PersistentAttributeInterceptable, SelfDirtinessTracker {
+
+	//TODO what about ExtendedSelfDirtinessTracker ?
```

Comment:
IMO `ExtendedSelfDirtinessTracker` should be merged into `SelfDirtinessTracker`. I don't see a value in having a separate "trait" for this. Also, I think we should deprecate the `inlineDirtyChecking` property and enable this by default to get rid of `SelfDirtinessTracker` in general. It was nice to have the option to disable this while it was new, but let's get over it and prepare to remove the config setting.

Reply:
Agreed - but you're looking at the patch version for 5.6. IMO for more substantial refactoring that belongs in `main`

Reply:
Of course, I just wanted to add these comments somewhere :)